main.go

```diff
@@ -104,31 +133,51 @@ func main() {
 		log.Fatalf("Failed to get instance of the helmrelease with %v", err)
 	} else if err != nil {
 		// This means that the object was not found
-		log.Infof("Did not find the helm release: %v, creating the release...", hr.Name)
-		// Create the HelmRelease if it doesn't exist
-		retry(ctx, func() error { return clientSet.Create(ctx, hr) }, interval)
+		log.Infof("Did not find the helm release: %v", hr.Name)
+		if action == Install {
```

Comment:
Can we refactor this whole thing into `deleteAction` and `installAction` so that we are not left with all these `if` statements and have potentially confusing code?

Reply:
Hope I understood your feedback correctly.. Now I split deleteAction and installAction from up top. Please take a look.